ipynb/1_diversity_analyses.ipynb

```diff
@@ -0,0 +1,623 @@
+    "  --m-metadata-file {map_file} \\\n",
+    "  --p-where \"Taxonomy_Order='Chiroptera'\" \\\n",
+    "  --o-filtered-table {wd}/picrust/cr_d5k.metagenome_predictions_bats.qza\n",
+    "\n",
+    "!qiime feature-table filter-samples \\\n",
+    "  --i-table {wd}/picrust/cr_d5k.metagenome_predictions.qza \\\n",
+    "  --m-metadata-file {map_file} \\\n",
+    "  --p-where \"Taxonomy_Genus='Geospiza'\" \\\n",
+    "  --o-filtered-table {wd}/picrust/cr_d5k.metagenome_predictions_birds.qza\n",
+    "\n",
+    "!qiime feature-table filter-samples \\\n",
+    "  --i-table {wd}/picrust/cr_d5k.metagenome_predictions.L3.qza \\\n",
+    "  --m-metadata-file {map_file} \\\n",
+    "  --p-where \"Taxonomy_Order='Chiroptera'\" \\\n",
+    "  --o-filtered-table {wd}/picrust/cr_d5k.metagenome_predictions_bats.L3.qza\n",
+    "\n",
+    "!qiime feature-table filter-samples \\\n",
+    "  --i-table {wd}/picrust/cr_d5k.metagenome_predictions.qza \\\n",
+    "  --m-metadata-file {map_file} \\\n",
+    "  --p-where \"Taxonomy_Genus='Geospiza'\" \\\n",
+    "  --o-filtered-table {wd}/picrust/cr_d5k.metagenome_predictions_birds.L3.qza"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## ANCOM for differential abundance testing"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#using unrarefied data\n",
+    "!qiime feature-table filter-samples \\\n",
+    "  --i-table {wd}/data/merged-table_metadata-matched.nochloro-nomito.qza \\\n",
+    "  --m-metadata-file {map_file} \\\n",
+    "  --p-where \"Taxonomy_Order='Chiroptera'\" \\\n",
+    "  --o-filtered-table {wd}/data/merged-table_metadata-matched.nochloro-nomito_bats.qza\n",
+    "    \n",
+    "!qiime feature-table filter-samples \\\n",
+    "  --i-table {wd}/data/merged-table_metadata-matched.nochloro-nomito.qza \\\n",
+    "  --m-metadata-file {map_file} \\\n",
+    "  --p-where \"Taxonomy_Genus='Geospiza'\" \\\n",
+    "  --o-filtered-table {wd}/data/merged-table_metadata-matched.nochloro-nomito_birds.qza"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#ancom works on log ratios, and thus zeroes must not be present. \n",
+    "#a pseudocount of 1 is added to everything\n",
+    "!mkdir {wd}/ancom\n",
+    "!qiime composition add-pseudocount \\\n",
+    "  --i-table {wd}/data/merged-table_metadata-matched.nochloro-nomito_bats.qza \\\n",
+    "  --o-composition-table {wd}/ancom/ancom_table_bats.qza\n",
+    "\n",
+    "!qiime composition add-pseudocount \\\n",
+    "  --i-table {wd}/data/merged-table_metadata-matched.nochloro-nomito_birds.qza \\\n",
+    "  --o-composition-table {wd}/ancom/ancom_table_birds.qza"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#!mkdir {wd}/picrust/ancom\n",
```

Comment:
Do you want to comment out this `!mkdir` line?

Reply:
Yeah, I could split up the notebooks in smaller blocks so that they could be run in order.  I'll work on that for the next commit. 
For the mkdir lines, I've commented out most of them so I wasn't rewriting directories for when I would rerun bits of the code. I need to go through and uncomment those.  Thanks for catching!

Reply:
Those lines are now uncommented out. 
Could you merge after reviewing?

Reply:
Sure thing! 

FYI for this purpose I like to us the following Python code:

```{python}
from os.path import makedirs

makedirs(out_dir, exist_ok=True)
```

This will make the directory if it doesn't exist, and also not complain if it does exist.